### ISSUE TITLE
RangeError: Maximum call stack size exceeded error if any of the custom attributes are named the same as HTMLElement keys

### ISSUE THREAD

**Issue by Maximillion195** [bitovi/react-to-webcomponent#68]
If you use an attribute named the same as any key properties in a HTMLElement interface. Then you get an infinite loop error `Uncaught RangeError: Maximum call stack size exceeded.`

For example if I use lang (https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/lang)

`<webcomponent lang="en"></ webcomponent>` 

Looks like it is this check that is the issue

```
      if (
        typeof key === "symbol" ||
        renderAddedProperties[key] ||
        key in target
      ) {
        return Reflect.set(target, key, value, receiver)
      } else {
        define.expando(receiver, key, value)
      }
```

It always sees 'lang' in target and keeps trying to set it.

https://codesandbox.io/s/react-to-webcomponent--basic-forked-ioepsn

**Comment by PtAuDev** [comment]
I also saw this in my project.  

There is a map that contains all the attributes of the component. After changing the condition logic, it works. Of course, now I use the source code instead of importing from the npm package.

```js
if (
typeof key === "symbol" ||
renderAddedProperties[key] ||
//attrPropMap is the collection of attributes
(key in target  && !Object.keys(attrPropMap).includes(key))
) {
        return Reflect.set(target, key, value, receiver)
} else {
        define.expando(receiver, key, value)
}
```

**Comment by christopherjbaker** [comment]
Hey, thanks for the bug @Maximillion195 and the fix @PtAuDev. I'll have somebody get on a patch for that asap, and pay close attention to this in v2 to see if there's a better approach here.

**Comment by christopherjbaker** [comment]
We just released v1.7.3 that should resolve this issue. We merged that fix into our 2.0 beta branch, but it is not in a state to be released at the moment.
